src/v3/src/components/Widget/index.tsx

```diff
@@ -386,23 +392,33 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
     if (isClientTransaction) {
       return;
     }
-    if (widgetRendered && typeof idxTransaction !== 'undefined'
-      && uischema.elements.length > 0) {
-      eventEmitter?.emit?.('afterRender', getEventContext(idxTransaction));
+    if (widgetRendered) {
+      (async () => {
+        const isFirstRender = !widgetRenderedOnce.current;
```

Comment:
```suggestion
        const isFirstRender = typeof widgetRenderedOnce.current === 'undefined';
```

Reply:
nit just to be more specific since its a ref

Reply:
or I believe you can just give the ref a default of `false` above so its set to a boolean